src/cache/redis/locking.js

```diff
@@ -0,0 +1,19 @@
+const { makeLockUnlockWithId, withLockByInput } = require("gamlajs").default;
+const { pipe, curryN } = require("ramda");
+
+const distributedLock = curryN(4, (argsToLockId, lockIdToKey, redisClient, f) =>
```

Comment:
as discussed please talk with @romCohen and understand why do we have a need for two different funcs here `argsToLockId` and `lockIdToKey`

Reply:
@kevinszuchet do we have an answer re this?

Reply:
"Basically you have 2 functions here, one to generate the lock id in redis. the second to get the redis key from the lockId"